extensions/amazon-aws/src/components/ecs/ECSClusterServiceTaskContainers.tsx

```diff
@@ -0,0 +1,109 @@
+import { ContainerDefinition } from "@aws-sdk/client-ecs";
+import { LogStartTimes } from "../../interfaces";
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import { fetchTaskContainers, getTaskContainerUrl } from "../../actions";
+import { getActionOpenInBrowser, getActionPush, getExportResponse, getFilterPlaceholder } from "../../util";
+import { useCachedPromise } from "@raycast/utils";
+import CloudwatchLogs from "../cloudwatch/CloudwatchLogs";
+import { useState } from "react";
+import CloudwatchLogsTimeDropdown from "../searchbar/CloudwatchLogsTimeDropdown";
+
+function ECSClusterServiceTaskContainers({ taskDefinitionArn }: { taskDefinitionArn: string }) {
+  const [logStartTime, setLogStartTime] = useState<LogStartTimes>(LogStartTimes.OneHour);
+  const { data: containers, isLoading } = useCachedPromise(fetchTaskContainers, [taskDefinitionArn], {
+    keepPreviousData: true,
+  });
+
+  const getActions = (container: ContainerDefinition) => {
+    const actionViewInApp = getActionPush({
+      title: "View Logs",
+      component: CloudwatchLogs,
+      logGroupName: container.logConfiguration?.options ? container.logConfiguration?.options["awslogs-group"] : "",
+      startTime: logStartTime,
+      logGroupStreamPrefix: container.logConfiguration?.options
+        ? container.logConfiguration?.options["awslogs-stream-prefix"]
+        : "",
+    });
+    const actionViewInBrowser = getActionOpenInBrowser(getTaskContainerUrl(taskDefinitionArn));
+
+    let containerActions = [actionViewInApp, actionViewInBrowser];
+
+    if (container.logConfiguration?.logDriver !== "awslogs") {
+      containerActions = containerActions.filter((action) => action.key !== "logs");
```

Comment:
I think this can be really difficult for someone to understand what is going on here. Relying on `key` for conditions seems quite fragile. I'm not even sure if this even works as I can't find `"logs"` as a key anywhere. I think in-place conditional rendering will work fine here:
In-place conditions will have better readability IMO:
```jsx
<ActionPanel>
  {container.logConfiguration?.logDriver === "awslogs" &&
    getActionPush({
      title: "View Logs",
      component: CloudwatchLogs,
      logGroupName: container.logConfiguration?.options ? container.logConfiguration?.options["awslogs-group"] : "",
      startTime: logStartTime,
      logGroupStreamPrefix: container.logConfiguration?.options
        ? container.logConfiguration?.options["awslogs-stream-prefix"]
        : "",
    })}
   {getActionOpenInBrowser(getTaskContainerUrl(taskDefinitionArn))}
```